### PR TITLE
More helpful error message in checkEnvironment.php

### PR DESCRIPTION
**Pull request by amosfolz**
Provide helpful error message in situation where `cache` `session` or `logs` is accidentally deleted or moved. 
The existing error was: 
image

This adds `checkDirectories` which provides a more helpful error message:
image

Because I was unable to come up with a more creative way to prevent `checkPermissions` from returning the generic error message, that check is skipped until the directories actually exist.

### CODE REVIEW

File: app/sprinkles/core/src/Util/CheckEnvironment.php [hunk 1]
```diff
@@ -125,6 +125,12 @@ public function checkAll()
             $problemsFound = true;
         }
 
+        if ($this->checkDirectories()) {
+            $problemsFound = true;
+        // Skip checkPermissions() if the required directories do not exist.
+        return $problemsFound;
```
Comment (lcharette) [inline app/sprinkles/core/src/Util/CheckEnvironment.php]: Careful with the indentation here

File: app/sprinkles/core/src/Util/CheckEnvironment.php [hunk 2]
```diff
@@ -257,6 +263,39 @@ public function checkPdo()
...
+        $problemsFound = false;
+
+        $directoryPaths = [
+            "logs"     => $this->locator->findResource('log://'),
+            "cache"    => $this->locator->findResource('cache://'),
+            "sessions" => $this->locator->findResource('session://')
+        ];
+
+        foreach ($directoryPaths as $directory => $path) {
+          if ($path == null) {
```
Comment (lcharette) [inline app/sprinkles/core/src/Util/CheckEnvironment.php]: Indentation should be 4 spaces

File: app/sprinkles/core/src/Util/CheckEnvironment.php [hunk 3]
```diff
@@ -257,6 +263,39 @@ public function checkPdo()
...
+            $problemsFound = true;
+            $this->resultsFailed['directory-' . $directory] = [
+                'title'   => "<i class='fa fa-file-o fa-fw'></i> A required directory was not found.",
+                'message' => "Please check that <code>userfrosting/app/$directory</code> exists.",
+                'success' => false
+            ];
+          }
+          else {
```
Comment (lcharette) [inline app/sprinkles/core/src/Util/CheckEnvironment.php]: Brace should be on the same line per PSR-2
https://www.php-fig.org/psr/psr-2/#51-if-elseif-else

**Comment by lcharette** [comment]
Related #934

File: app/sprinkles/core/src/Util/CheckEnvironment.php (hunk 1 above)
Comment (amosfolz) [reply app/sprinkles/core/src/Util/CheckEnvironment.php]: Thanks for the information! I will look over the PSR guides and get more familiar with the UF style guide as well.

**Comment by lcharette** [comment]
Merged into hotfix for 4.2.1 release. Thanks !

**Pull request merged**